pkg/controller/nodes/task/handler.go

```diff
@@ -536,39 +537,47 @@ func (t Handler) Handle(ctx context.Context, nCtx handler.NodeExecutionContext)
 	// STEP 1: Check Cache
 	if (ts.PluginPhase == pluginCore.PhaseUndefined || ts.PluginPhase == pluginCore.PhaseWaitingForCache) && checkCatalog {
 		// This is assumed to be first time. we will check catalog and call handle
-		entry, err := t.CheckCatalogCache(ctx, tCtx.tr, nCtx.InputReader(), tCtx.ow)
-		if err != nil {
-			logger.Errorf(ctx, "failed to check catalog cache with error")
-			return handler.UnknownTransition, err
-		}
-
-		if entry.GetStatus().GetCacheStatus() == core.CatalogCacheStatus_CACHE_HIT {
-			r := tCtx.ow.GetReader()
-			if r == nil {
-				return handler.UnknownTransition, errors.Errorf(errors.IllegalStateError, nCtx.NodeID(), "failed to reader outputs from a CacheHIT. Unexpected!")
-			}
-
-			// TODO @kumare this can be optimized, if we have paths then the reader could be pipelined to a sink
-			o, ee, err := r.Read(ctx)
+		// If the cache should be skipped (requested by user for the execution), do not check datacatalog for any cached
+		// data, but instead always perform calculations again and overwrite the stored data after successful execution.
+		if nCtx.ExecutionContext().GetExecutionConfig().SkipCache {
+			logger.Info(ctx, "Execution config forced cache skip, not checking catalog")
+			pluginTrns.PopulateCacheInfo(catalog.NewCatalogEntry(nil, cacheSkipped))
```

Comment:
Just to make sure, we are setting the cache status to `CACHE_SKIPPED` here during the cache lookup, but this will be overwritten by a `CACHE_POPULATED` once the data is written. So really this functionality is overwritting the existing cache and skipping just the cache lookup right? I wonder if it makes sense to update the flag name. I'm worried that `skipCache` might be understood as skipping both the cache lookup and write, like executing without the cache involved, rather than just skipping the cache lookup. Thoughts?

Reply:
That's correct. I only noticed the status gets overwritten after implementing the new `CACHE_SKIPPED` enum value, but decided to keep it anyway in case we need to check the status at some other point internally. I believe it's actually never returned to a client at the moment since the cache will either be written (thus returning `CACHE_POPULATED`) or some error occurs.

I was unsure about the best naming for this functionality as well, I believe my original naming in the RFC was `evict_cache`, but I agree it's a bit more than actually just _skipping_ the cached entry. Happy to rename it to `overwriteCache` as that's clearer about what exactly will happen (and the internal flags are already called `overwrite` in some places anyways :sweat_smile: ).